### PR TITLE
fix/V1-143 - Progress bar on details page

### PR DESCRIPTION
**Pull request by zhuk-vezunchikz**

### CODE REVIEW

**Review by k-krasilnikova — CHANGES_REQUESTED** [review]

File: frontend/src/pages/DetailedCourse/DetailedCourseContainer.tsx [hunk 1]
```diff
@@ -7,6 +7,8 @@ import { getWindowWidth } from 'utils/helpers/getWindowWidth';
 import { COURSE_STATUSES } from 'constants/statuses';
 
 import DetailedCourse from './DetailedCourse';
+import { COMPLETED_STATUS_TEXT, FAILED_STATUS_TEXT } from '../../constants/detailedCourse';
```
Comment (irina-sinitsyna) [inline frontend/src/pages/DetailedCourse/DetailedCourseContainer.tsx:10]: Please use absolute paths here for constants and fix imports order after it

**Review by irina-sinitsyna — CHANGES_REQUESTED** [review]

File: frontend/src/pages/DetailedCourse/DetailedCourseContainer.tsx [hunk 2]
```diff
@@ -62,6 +64,40 @@ const DetailedCourseContainer: React.FC<Props> = ({ page }) => {
...
+        break;
+      case COURSE_STATUSES.completed:
+        progressValue = 100;
+        progressText = COMPLETED_STATUS_TEXT;
+        progressVariant = VARIANTS.completed;
+        break;
+      case COURSE_STATUSES.failed:
+        progressValue = 0;
+        progressText = FAILED_STATUS_TEXT;
+        progressVariant = VARIANTS.failed;
```
[the remark this answers is not in the document]
Comment (zhuk-vezunchikz) [reply frontend/src/pages/DetailedCourse/DetailedCourseContainer.tsx]: done

File: frontend/src/pages/DetailedCourse/DetailedCourseContainer.tsx [hunk 3]
```diff
@@ -4,11 +4,10 @@ import { useParams } from 'react-router';
 import { useApplyCourse, useGetCourseInfo } from 'api/courses';
 import { useGetClientCourseInfo } from 'api/myCourses';
 import { getWindowWidth } from 'utils/helpers/getWindowWidth';
+import convertStatusToProgress from 'utils/helpers/convertStaatusToProgress';
```
Comment (irina-sinitsyna) [inline frontend/src/pages/DetailedCourse/DetailedCourseContainer.tsx:7]: `import convertStatusToProgress from 'utils/helpers/convertStatusToProgress';`

Comment (zhuk-vezunchikz) [reply frontend/src/pages/DetailedCourse/DetailedCourseContainer.tsx]: done

**Review by irina-sinitsyna — DISMISSED** [review]

**Review by irina-sinitsyna — DISMISSED** [review]

**Review by Julia-Maliutsina — DISMISSED** [review]

File: frontend/src/pages/DetailedCourse/DetailedCourseContainer.tsx [hunk 4]
```diff
@@ -62,6 +63,17 @@ const DetailedCourseContainer: React.FC<Props> = ({ page }) => {
...
+  let progressValue;
+  let progressText;
+  let progressVariant;
+
+  if (courseData) {
+    const progress = convertStatusToProgress(courseData.status);
+    progressValue = progress.progressValue;
+    progressText = progress.progressText;
+    progressVariant = progress.progressVariant;
+  }
```
Comment (k-krasilnikova) [inline frontend/src/pages/DetailedCourse/DetailedCourseContainer.tsx:75]: Move this check to the useEffect, it will help us to avoid an additional render

**Review by k-krasilnikova — CHANGES_REQUESTED** [review]

**Pull request merged**